File: src/client/python/agentguard/checkers/llm_after/llm_output.py
"""Checker for LLM output events."""
from __future__ import annotations

from agentguard.checkers.base import BaseChecker, CheckResult
from agentguard.checkers.common.patterns import find_signals, text_of
from agentguard.checkers.registry import register
from agentguard.schemas.context import RuntimeContext
from agentguard.schemas.events import EventType, RuntimeEvent


@register(
    name="llm_output",
    description="Detect risky content, secrets, and injection patterns in LLM output.",
)
class LLMOutputChecker(BaseChecker):
    event_types = [EventType.LLM_OUTPUT]

    def check(self, event: RuntimeEvent, context: RuntimeContext) -> CheckResult:
        text = text_of(event.payload.get("output"))
        return CheckResult(risk_signals=find_signals(text))

File: src/client/js/agentguard/checkers/llm_after/llm_output.js
"use strict";

const { BaseChecker, CheckResult } = require("../base");
const { EventType } = require("../../schemas/events");
const { matchSignals } = require("../common/patterns");

class LLMOutputChecker extends BaseChecker {
  constructor() {
    super();
    this.event_types = [EventType.LLM_OUTPUT];
  }

  check(event) {
    const text = JSON.stringify(event.payload || {});
    return new CheckResult({ risk_signals: matchSignals(text) });
  }
}

module.exports = {
  LLMOutputChecker,
};
